sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -510,6 +538,21 @@ class SessionCatalog(
     externalCatalog.listPartitions(db, table, partialSpec)
   }
 
+  /**
+   * Verify if the input partition spec exactly matches the existing defined partition spec
+   * The columns must be the same but the orders could be different.
+   */
+  private def requireExactMatchedPartitionSpec(
```

Comment:
I think the interface here could be better, maybe something like
```
private def requirePartitionsDefinedInTable(
      specs: Seq[TablePartitionSpec],
      table: CatalogTable): Unit = {
  ...
}
```
This would allow you to reuse more code